src/memory/cgroup.rs

```diff
@@ -0,0 +1,38 @@
+use std::path::Path;
+
+use super::Memory;
+use crate::{dir_exists, ProbeError, Result};
+
+/// Read the current memory status of the container.
+#[cfg(target_os = "linux")]
+pub fn read_from_container() -> Result<Memory> {
```

Comment:
```suggestion
pub fn read() -> Result<Memory> {
```

As suggested in https://github.com/appsignal/appsignal-agent/pull/945#discussion_r1106798913